azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CAuthorizationRequestResolver.java

```diff
@@ -17,42 +22,67 @@
 
 public class AADB2CAuthorizationRequestResolver implements OAuth2AuthorizationRequestResolver {
 
-    private final AADB2CProperties properties;
+    private static final String REQUEST_BASE_URI =
+            OAuth2AuthorizationRequestRedirectFilter.DEFAULT_AUTHORIZATION_REQUEST_BASE_URI;
+
+    private static final String REGISTRATION_ID_NAME = "registrationId";
+
+    private static final String MATCHER_PATTERN = String.format("%s/{%s}", REQUEST_BASE_URI, REGISTRATION_ID_NAME);
+
+    private static final AntPathRequestMatcher requestMatcher = new AntPathRequestMatcher(MATCHER_PATTERN);
 
     private final OAuth2AuthorizationRequestResolver defaultResolver;
 
-    public AADB2CAuthorizationRequestResolver(ClientRegistrationRepository registrationRepository,
-                                              AADB2CProperties properties) {
-        this.defaultResolver = new DefaultOAuth2AuthorizationRequestResolver(registrationRepository,
-                "/oauth2/authorization");
-        // TODO(pan): Investigate why should be below string.
-        this.properties = properties;
+    public AADB2CAuthorizationRequestResolver(@NonNull ClientRegistrationRepository repository) {
+        this.defaultResolver = new DefaultOAuth2AuthorizationRequestResolver(repository, REQUEST_BASE_URI);
     }
 
     @Override
-    public OAuth2AuthorizationRequest resolve(HttpServletRequest request) {
-        Assert.notNull(request, "request cannot be null.");
+    public OAuth2AuthorizationRequest resolve(@NonNull HttpServletRequest request) {
+        if (requestMatcher.matches(request)) {
+            return getB2CAuthorizationRequest(defaultResolver.resolve(request), getRegistrationId(request));
+        }
 
-        return getAADB2CAuthorizationRequest(defaultResolver.resolve(request));
+        return null;
```

Comment:
Returning null is not a good practice

Reply:
yes, but I have to follow the implicit rule with spring security.

Reply:
Will add comments for this.